### DISCUSSION TITLE
Make sense to have one Redis and multiple Sidekiq processes on different db's

### DISCUSSION THREAD

**Discussion by cavi21** [sidekiq/sidekiq#5101] [Q&A] [answered]
Hi there! I search for this on the issues and in here, but couldn't find an anwser, sorry if this was already asked.

I was wondering if there is something against (or it's not a good practice) to have **one** Redis instance (planning to have Redis Sentinel setup) and different apps that each one have their own sidekiq process, but each connects to the same Redis but on different DBs.

So imagine that we have the Redis on `redis://redis.company.test:6379` and we have different apps, so in each of them we have sidekiq configured like:
 - App 1: `config.redis = { url: "redis://redis.company.test:6379/1" }`
 - App 2: `config.redis = { url: "redis://redis.company.test:6379/2" }`
 - App 3: `config.redis = { url: "redis://redis.company.test:6379/3" }`

Is this ok?, or it could generate some issue that I'm not seeing or being aware of?

Thanks!

**Comment by mperham** [comment, marked as answer]
It’s not recommended as there are limitations. The apps need to use the
same versions of Sidekiq and all apps will share Redis’s single thread and
so will not scale independently.

**Reply by cavi21** [reply]
Hi @mperham thanks for the anwser and clarification! is not the way to go then :)!

**Reply by cavi21** [reply]
I kept thinking that It's weird to have the possibility os using different dbs but with not much gains, but maybe there are some use cases for them, I understand that's a Redis thing, but well!

**Reply by pdfrod** [reply]
I'm curious about why all apps need to be running the same Sidekiq version. I can see that there could be problems if we use Pub/Sub (since Redis Pub/Sub channels are not scoped to a single database), but if we don't use Pub/Sub what other issues could there be?

**Reply by mperham** [reply]
@pdfrod you're right, I can't think of why the versions must be identical so I'm not sure what scenario I was thinking about. That would be true if they were sharing the same DB but if they are using different DBs then the data should be isolated per app.

**Reply by pdfrod** [reply]
OK, thanks for the answer!

**Reply by mmustala** [reply]
Now that the pidfile options are gone, how can I know which process to send signals when deploying? I'm currently on 5.2.7 and running multiple Sidekiqs for different apps on one server.

**Reply by mperham** [reply]
You use systemd to manage services and systemd keeps track of the process.

**Reply by mmustala** [reply]
Ah, I just need to create a systemd file for each of my services. I can call them sidekiq_app1, sidekiq_app2 etc.

**Reply by cavi21** [reply]
Hi @mmustala you can create only 1 file as `sidekiq@.service` and then you can manage all the services with `systemctl start sidekiq@app1.service`, `systemctl start sidekiq@app2.service`, etc. And you can access the values `app1`, `app2` within the systemd file with `%i`. There are several places to look into more detail, here is one https://opensource.com/article/20/12/multiple-service-instances-systemctl

Hope this helps
